idgenerator.go

```diff
@@ -0,0 +1,43 @@
+package eventsourcing
+
+import "crypto/rand"
+
+// idFunc is a global function that generates aggregate id's.
+// It could be changed from the outside via the SetIDFunc function.
+var idFunc = randSeq
+
+// SetIDFunc is used to change how aggregate ID's are generated
+// default is a random string
+func SetIDFunc(f func() string) {
+	idFunc = f
+}
+
+func randSeq() string {
+	id, err := generateRandomString(32)
+	if err != nil {
+		return ""
+	}
+	return id
+}
+
+func generateRandomString(n int) (string, error) {
+	const letters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-"
+	bytes, err := generateRandomBytes(n)
+	if err != nil {
+		return "", err
+	}
+	for i, b := range bytes {
+		bytes[i] = letters[b%byte(len(letters))]
+	}
+	return string(bytes), nil
+}
+
+func generateRandomBytes(n int) ([]byte, error) {
+	b := make([]byte, n)
+	_, err := rand.Read(b)
+	// Note that err == nil only if we read len(b) bytes.
+	if err != nil {
```

Comment:
Is it OK to simply have `return b, err`? 

If `err` is not nil it implies that `b` is unsafe to use.
```suggestion
	return b, err
```